python/elasticbeanstalk/elasticbeanstalk-bg-pipeline/resources/blue_green.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import print_function
+from boto3.session import Session
+
+import json
+import urllib
+import boto3
+import zipfile
+import tempfile
+import botocore
+import traceback
+import time
+
+print('Loading function')
+
+eb = boto3.client('elasticbeanstalk')
+code_pipeline = boto3.client('codepipeline')
+
+def put_job_success(job, message):
+    """Notify CodePipeline of a successful job
+
+    Args:
+        job: The CodePipeline job ID
+        message: A message to be logged relating to the job status
+
+    Raises:
+        Exception: Any exception thrown by .put_job_success_result()
+
+    """
+    print('Putting job success')
+    print(message)
+    code_pipeline.put_job_success_result(jobId=job)
+
+def put_job_failure(job, message):
+    """Notify CodePipeline of a failed job
+
+    Args:
+        job: The CodePipeline job ID
+        message: A message to be logged relating to the job status
+
+    Raises:
+        Exception: Any exception thrown by .put_job_failure_result()
+
+    """
+    print('Putting job failure')
+    print(message)
+    code_pipeline.put_job_failure_result(jobId=job, failureDetails={'message': message, 'type': 'JobFailed'})
+
+def get_user_params(job_data):
+    """Decodes the JSON user parameters and validates the required properties.
+
+    Args:
+        job_data: The job data structure containing the UserParameters string which should be a valid JSON structure
+
+    Returns:
+        The JSON parameters decoded as a dictionary.
+
+    Raises:
+        Exception: The JSON can't be decoded or a property is missing.
+
+    """
+    try:
+        # Get the user parameters which contain the stack, artifact and file settings
+        user_parameters = job_data['actionConfiguration']['configuration']['UserParameters']
+        decoded_parameters = json.loads(user_parameters)
+
+    except Exception as e:
```

Comment:
You are not using `e` anywhere.  Also, at the moment you are catching ALL exceptions here but only raising an Exception if one of your required parameters is missing.  All other exceptions will be swallowed here.  That doesn't seem right.